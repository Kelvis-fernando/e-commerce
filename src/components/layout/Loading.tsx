import { CircleNotch } from "phosphor-react";
import { LoadingContainer, LoadingContent } from "../../styles/layout/loading";

interface LoagingProps {
  message?: string;
}

const Loading = ({ message = "Carregando..." }: LoagingProps) => {
  return (
    <LoadingContainer>
      <LoadingContent>
        <CircleNotch size={25} />
        <h1>{message}</h1>
      </LoadingContent>
    </LoadingContainer>
  );
};

export default Loading;
