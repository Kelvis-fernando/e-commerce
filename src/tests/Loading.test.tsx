import React from "react";
import { render } from "@testing-library/react";
import Loading from "../components/layout/Loading";

describe("Loading", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<Loading />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
