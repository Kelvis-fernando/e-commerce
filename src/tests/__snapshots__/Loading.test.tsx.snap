// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe dVEnEb"
  >
    <div
      class="sc-gswNZR eUScaB"
    >
      <svg
        fill="currentColor"
        height="25"
        viewBox="0 0 256 256"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <path
          d="M168,40.7a96,96,0,1,1-80,0"
          fill="none"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
      <h1>
        Carregando...
      </h1>
    </div>
  </div>
</div>
`;
